SS 473 chosen flavor is not pre selected

sandstromviktor opened the pull request:
## Description

Sorry the change in many files.... 

I've done the following:
1. Added `flavor` to `Appinstance`. This allows us to always display the flavor that is being used. It displays the flavor name and `(active)` to show that it's in use. 
2. I've fixed the same issue with the persistent volume. This one was complicated since you must fetch the mounted volume through the json parameters. It seems to work pretty good. 
3. I removed the unnecessary field `repository` from project 
4. Fixed the unit and e2e tests to match the changes.

When i changed the migration files to add flavor to appinstance, i decided to clean up the migration files. This will make the future model changes a bit easier i believe. 

Let me know if you have questions. 

- [x] E2E passes
- [x] Unit tests passes


## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [ ] I have included a link to the issue on GitHub or JIRA (if any)
- [x] I have included migration files (if there are changes to the model classes)
- [x] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [ ] I have updated the release notes (releasenotes.md)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

sandstromviktor commented on apps/models.py:
```diff
@@ -164,6 +164,12 @@ class AppInstance(models.Model):
         on_delete=models.CASCADE,
         related_name="appinstance",
     )
+    flavor = models.ForeignKey(
+        "projects.Flavor",
+        on_delete=models.CASCADE,
+        related_name="appinstance",
+        null=True,
+    )
```
This adds flavor to `Appinstance`. I added `on_delete` condition so if we remove a flavor from a project, it will automatically remove the appinstances that have that flavor.

sandstromviktor commented on projects/models.py:
```diff
@@ -259,15 +257,13 @@ class Project(models.Model):
     project_key = models.CharField(max_length=512)
     project_secret = models.CharField(max_length=512)
     # ----------------------
-    # These fields should be removed.
-    repository = models.CharField(max_length=512, null=True, blank=True)
-    repository_imported = models.BooleanField(default=False)
-    # ----------------------
```
Removed these

sandstromviktor commented on apps/migrations/0002_initial.py:
```diff
@@ -63,6 +63,11 @@ class Migration(migrations.Migration):
+        migrations.AddField(
+            model_name='appinstance',
+            name='flavor',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.RESTRICT, related_name='appinstance', to='projects.flavor'),
+        ),
```
There is a hierarchy:
Appinstance ---has---> Flavor ---has---> Project.
I've added a restriction so that deleting a flavor that is used by an appinstance is not permitted. However, deleting a project, is allowed because it deletes both the flavor and appinstance.

hamzaimran08 replied:
Yes, the PR looks good now. I haven't had time to test it manually though.

churnikov commented on studio/migrations/projects/0001_initial.py:
```diff
@@ -168,4 +171,24 @@ class Migration(migrations.Migration):
             name='project',
             field=models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, related_name='ba_project', to='projects.project'),
         ),
+        migrations.AddField(
+            model_name='environment',
+            name='public',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='projecttemplate',
+            name='enabled',
+            field=models.BooleanField(default=True),
+        ),
+        migrations.AlterField(
+            model_name='environment',
+            name='project',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='projects.project'),
+        ),
+        migrations.AlterField(
+            model_name='environment',
+            name='slug',
+            field=models.CharField(blank=True, max_length=100, null=True),
+        ),
```
I wonder if this will work on prod db

sandstromviktor replied:
I think this is unchanged migrations. Why wouldn't it work you think?

sandstromviktor commented:
> LGTM
> 
> Tried on my local machine, look nice
> 
> 
> 
> Couple of question though
> 
> 1. What are these deletes for?
> 
> <img width="962" alt="image" src="https://github.com/ScilifelabDataCentre/stackn/assets/8545082/b45b2ea1-d75e-44e7-bced-40a50c9c5f60">
> 
> 
> 
> 2. Why so many migrations are deleted?
> 
> <img width="847" alt="image" src="https://github.com/ScilifelabDataCentre/stackn/assets/8545082/0315a6b2-9eb5-4e0e-a753-0cb4745eee54">
> 
> 

1. The deletes are just a clean up of a model attribute that is not used but needs to be set every time we create projects in tests etc. Just unnecessary garbage. 

2. All deleted migration were autogenerated by Django, but since we are starting with a fresh database, I cleaned up all the mess that were there. A bunch of fields being added, changed, removed and added again etc.

churnikov commented on apps/helpers.py:
```diff
@@ -9,6 +9,7 @@
 from .serialize import serialize_app
 from .tasks import deploy_resource
 
+from projects.models import Flavor
 ReleaseName = apps.get_model(app_label=settings.RELEASENAME_MODEL)
```
Seems like you didn't apply pre-commit hooks, it would have formatted this

sandstromviktor replied:
Done!

churnikov commented on apps/models.py:
```diff
@@ -164,6 +164,12 @@ class AppInstance(models.Model):
+        "projects.Flavor",
+        on_delete=models.RESTRICT,
+        related_name="appinstance",
+        null=True,
```
Why do you think that it could be null?

sandstromviktor replied:
Removed that now!
